### PR TITLE
Create KafkaSources OIDC service account and expose in its status

### LINKED ISSUE #3658: [control-plane]: Support auto generation of KafkaSource identity service account and expose in `AuthStatus`
**Problem** 

As the [Eventing OIDC feature track](https://docs.google.com/document/d/1e7UgNTkL0Br5Da09Rg2ieVmhKJo4VXuBj-mHT9NCujY/edit#bookmark=id.gvbj3362t2fy) describes, the `AuthStatus` is meant to provide the generated service account name in the resource status.

When https://github.com/knative/eventing/issues/7173 is done, we should:
* when the feature flag (see https://github.com/knative/eventing/issues/7174) is disabled:
  * no change in behavior
* when the feature flag (see https://github.com/knative/eventing/issues/7174) is enabled:
  * create the service account for the KafkaSources identity
  * expose the service accounts name in `.status.auth.serviceAccountName`

**Additional Information:**
* See implementation for trigger in eventing-core: https://github.com/knative/eventing/issues/7222

### PR DESCRIPTION
**Pull request by creydr**
Fixes #3658

## Proposed Changes

- :gift: Create KafkaSources OIDC service account and expose in its status

**Release Note**

```release-note
Create KafkaSources OIDC service account and expose in its status
```

### CODE REVIEW

**Review by Leo6Leo — COMMENTED** [review]
@creydr  I don't see any issues in the code itself except for some linter error

**Comment by creydr** [comment]
> @creydr I don't see any issues in the code itself except for some linter error

updated :) Can you recheck?

File: control-plane/pkg/reconciler/source/controller.go [hunk 1]
```diff
@@ -34,29 +37,51 @@ import (
...
+
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	serviceaccountinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/serviceaccount"
 )
 
 func NewController(ctx context.Context, watcher configmap.Watcher) *controller.Impl {
 
 	kafkaInformer := kafkainformer.Get(ctx)
 	consumerGroupInformer := consumergroupinformer.Get(ctx)
+	serviceaccountInformer := serviceaccountinformer.Get(ctx)
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/source/controller.go:49]: Should we already use the filtered informer ?

**Review by pierDipi — APPROVED** [review]
/lgtm

/hold for the comment

File: control-plane/pkg/reconciler/source/controller.go (hunk 1 above)
Comment (creydr) [reply control-plane/pkg/reconciler/source/controller.go:49]: We have #3655, where we do this for multiple. So I would do this in the other PR :shrug:

Comment (creydr) [reply control-plane/pkg/reconciler/source/controller.go:49]: /unhold

**Pull request merged**